raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -258,37 +265,29 @@ private void updateFollowerHighWatermark(FollowerState state, OptionalLong highW
     }
 
     private void updateLeaderEndOffsetAndTimestamp(LeaderState state) {
-        if (state.updateLocalEndOffset(log.endOffset())) {
+        if (state.updateLocalState(time.milliseconds(),
```

Comment:
nit: either use one line per parameter, or just one line for the whole call. Ditto below.

Reply:
@abbccdda raised a point that we only decide if it is the voter or not inside updateReplicaState, so the second part of the above comment actually would not work.

Thinking about it a bit more, may be we can just modify the semantics of the fetch timeout on leader to, when there’s no hw advances within the elapsed time, instead of when there’s no fetch from majority of the voters within the elapsed time? That has a similar indication of the “progress” and also we can just rely on the returned boolean from updateReplica/LocalState to decide if timer can be reset.